frontend/src/components/shared/VirtualDropdown/VirtualDropdown.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { ReactElement } from "react"
+import { StyledDropdownListItem } from "baseui/select"
+import { StyledList, StyledEmptyState, OptionListProps } from "baseui/menu"
+import { FixedSizeList } from "react-window"
+
+const LIST_ITEM_HEIGHT = 40
+const EMPTY_LIST_HEIGHT = 90
+const MAX_LIST_HEIGHT = 300
+
+interface FixedSizeListeItemProps {
+  data: { props: OptionListProps }[]
+  index: number
+  style: React.CSSProperties
+}
+
+function FixedSizeListItem(props: FixedSizeListeItemProps): ReactElement {
+  const { data, index, style } = props
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const { item, overrides, ...restChildProps } = data[index].props
+  return (
+    <StyledDropdownListItem
+      key={item.value}
+      style={{
+        boxSizing: "border-box",
+        paddingTop: 0,
+        paddingBottom: 0,
+        display: "flex",
+        alignItems: "center",
+        ...style,
+      }}
+      {...restChildProps}
+    >
+      {item.label}
+    </StyledDropdownListItem>
+  )
+}
+
+const VirtualDropdown = React.forwardRef((props: any, ref) => {
```

Comment:
Would love a comment for posterity about why this exists at all! i.e. calling out that this makes HUGE lists performant (do we even have an upper bound anymore?), and also prevents them from filling up the page =)